src/civetweb.c

```diff
@@ -5094,6 +5094,11 @@ connect_socket(struct mg_context *ctx /* may be NULL */,
 {
 	int ip_ver = 0;
 	*sock = INVALID_SOCKET;
+	
+	if (sa == NULL) {
+		return 0;
```

Comment:
The function declaration already requires sa to be not null (Line 5092):

      union usa *sa /* output: socket address, must not be NULL  */